### PR TITLE
Resolve: Add option to configure key caching duration (#24)

### PR DESCRIPTION
**Pull request by Foxcool**
This PR is just adds cacheDuration parameter. When specified, the default value (15 minutes) is replaced.

### CODE REVIEW

**Comment by Foxcool** [comment]
Also 1 test in master is failed

=== RUN   TestJWK
--- FAIL: TestJWK (0.00s)
    jwk_test.go:15: open cert.pem: no such file or directory


is it ok?

File: jws.go [hunk 1]
```diff
@@ -18,18 +19,19 @@ const (
...
-	RolesKey           string   `json:"roles_key,omitempty"`
-	CookieKey          string   `json:"cookie_key,omitempty"`
-	CipherSuites       []uint16 `json:"cipher_suites,omitempty"`
-	DisableJWKSecurity bool     `json:"disable_jwk_security"`
-	Fingerprints       []string `json:"jwk_fingerprints,omitempty"`
-	LocalCA            string   `json:"jwk_local_ca,omitempty"`
+	Alg                string        `json:"alg"`
+	URI                string        `json:"jwk-url"`
+	CacheEnabled       bool          `json:"cache,omitempty"`
+	CacheDuration      time.Duration `json:"cacheDuration,omitempty"`
```
Comment (taik0) [inline jws.go]: The json name should be `cache_duration`

**Comment by kpacha** [comment]
@Foxcool I think this is a great idea and I'm loving you make it backward compatible. On the other side, I think the TTL should not be too long in order to properly react to a rotation of the keys

regarding your comment, there is a test that requires a cert and key. They are generated with this line from the `jwk_test.go`:

```
//go:generate go run $GOROOT/src/crypto/tls/generate_cert.go --rsa-bits 1024 --host 127.0.0.1,::1,localhost --ca --start-date "Jan 1 00:00:00 1970" --duration=1000000h
```

the CI config calls `go generate` before launching the test suite

https://github.com/devopsfaith/krakend-jose/blob/master/.travis.yml#L11-L13

File: jws.go (hunk 1 above)
Comment (Foxcool) [reply jws.go]: Oh. Yes.

**Comment by Foxcool** [comment]
I accept a duration value of less than a second. Maybe I need to increase the step to a second?

**Comment by kpacha** [comment]
I'd change the type of the new property and instead of using a `time.Duration` I'd go with a plain `int` representing minutes.

otherwise, the duration will be defined in nanoseconds (https://play.golang.org/p/Hw0QJ7XDgdE)

**Comment by Foxcool** [comment]
> I'd change the type of the new property and instead of using a `time.Duration` I'd go with a plain `int` representing minutes.
> 
> otherwise, the duration will be defined in nanoseconds (https://play.golang.org/p/Hw0QJ7XDgdE)

Yes, this is what I meant in the previous comment. A nanosecond is too small a step. But the minute, it seems to me, is too large. Maybe a second is better?

**Comment by kpacha** [comment]
ouch, I did not understand you... 😄 

I think it's ok to use seconds instead of minutes.

**Comment by kpacha** [comment]
thanks for the contribution!

**Pull request merged**